src/ray/core_worker/core_worker.cc

```diff
@@ -1273,6 +1274,31 @@ bool CoreWorker::AddActorHandle(std::unique_ptr<ActorHandle> actor_handle,
   reference_counter_->AddLocalReference(actor_creation_return_id, CurrentCallSite());
   direct_actor_submitter_->AddActorQueueIfNotExists(actor_id);
 
+  // If actor handle is not persisted to GCS yet, we should subscribe
+  // worker failure events to figure out if actor is dead or not.
+  // SANG-TODO Add a link to the issue.
+  if (!actor_handle->IsPersistedToGCS()) {
+    // Call a function instead and test it.
+    rpc::Address owner_address = actor_handle->GetOwnerAddress();
+    WorkerID worker_id =
+        WorkerID::FromBinary(actor_handle->GetOwnerAddress().worker_id());
+    ActorID actor_id = actor_handle->GetActorID();
+    // SANG-TODO delete this.
+    RAY_LOG(ERROR) << "Location is not resolved Yet! actor id: " << actor_id
+                   << " owner worker id: " << worker_id;
+    auto on_worker_failure = [this, actor_id, worker_id](
+                                 const WorkerID &id,
+                                 const gcs::WorkerFailureData &worker_failure_data) {
+      RAY_LOG(ERROR) << "Worker failure!!";
+      if (id == worker_id) {
+        direct_actor_submitter_->DisconnectActor(actor_id, true);
+      }
+      // SANG-TODO: Should unsubscribe.
+    };
+    RAY_CHECK_OK(gcs_client_->Workers().AsyncSubscribeToWorkerFailures(
```

Comment:
Instead of subscribing, it'd be better to do a lookup to the worker failure table to check whether that worker has been reported as dead. You may have to add this functionality though.

Make sure to also check the node table (the worker's failure will not get reported if it's because the worker's raylet died). You can do that with `gcs_client_->Nodes().Get(node_id)`. And it'd be good to make sure that the test cases cover both scenarios :)

Reply:
Actually, I have one question! What's the benefit of using lookup instead of subscription?  (given I will subscribe only one entry instead of subscribe all entries!)

Reply:
Actually, nvm! It should be because we need to also check node status. I will use lookup command instead!